App.js

```diff
@@ -11,42 +11,55 @@ const chevronStyle = {width: 30, height: 30, resizeMode: 'contain', marginRight:
 const chatTitleStyle = {fontSize: 18, fontWeight: 'bold'};
 const titleStyle = {fontSize: 24, fontWeight: 'bold', flex: 1};
 
+const keysBlocklist = ['key', 'stale', 'routeNames']
+
+const stripNavigationState = (state) => {
+  const rawState = Array.isArray(state) ? [] : {};
+  for (const key of Object.keys(state)) {
+    if (keysBlocklist.includes(key)) {
+      continue
+    }
+    rawState[key] = typeof state[key] === 'object' ? stripNavigationState(state[key]) : state[key]
+  }
+  return rawState 
+}
+
 const config = {
-  initialRouteName: '',
+  initialRouteName: 'LeftHandNav',
   screens: {
     Chat: 'r/:id?',
     LeftHandNav: '',
     SettingsStack: {
+      initialRouteName: 'Settings',
+      path: 'settings',
       screens: {
-        Settings: 'settings',
-        About: 'settings/about',
+        Settings: '',
+        About: 'about',
       },
     },
     Search: 'search'
   },
 };
 
+
+const addChatRouteIfNecessary = (state) => {
+    if (!_.find(state.routes, r => r.name === 'Chat')) {
```

Comment:
Based on what Marc pointed out [here](https://github.com/marcaaron/navigation-test/pull/8/files#r1054838939) I think we can conditionally add a `chat` screen.
```js
if(path === '/' && !checkIsSmallScreen(Dimensions.get('window').width)) {
      addChatRouteIfNecessary(state)
}
```

Reply:
maybe because it is exposing an implementation details that we are "adding a chat route" and also it "might not happen". So, a more general name like `fixState()` could be OK and if you need to know what that means you go look at the function.